src/layers.c

```diff
@@ -131,22 +145,36 @@ surface_commit_notify(struct wl_listener *listener, void *data)
 		return;
 	}
 
+	struct output *output = (struct output *)layer_surface->output->data;
 	if (layer_surface->current.committed
 			|| layer->mapped != layer_surface->mapped) {
 		layer->mapped = layer_surface->mapped;
-		struct output *output =
-			output_from_wlr_output(layer->server, wlr_output);
 		layers_arrange(output);
 	}
+
+	/*
+	 * We need to call cursor_update_focus() to get a pointer-enter event
+	 * when a new layer surface appears under the cursor.  This enables
+	 * gdk_device_get_window_at_position() to work without moving pointer.
+	 * See issue #667
+	 */
+	cursor_update_focus(output->server);
 }
 
 static void
 unmap(struct lab_layer_surface *layer)
 {
 	struct seat *seat = &layer->server->seat;
-	if (seat->focused_layer == layer->scene_layer_surface->layer_surface) {
+	struct wlr_layer_surface_v1 *layer_surface =
+		layer->scene_layer_surface->layer_surface;
+
+	if (seat->focused_layer == layer_surface) {
 		seat_set_focus_layer(seat, NULL);
 	}
+	if (seat->pressed.surface == layer_surface->surface) {
+		seat->pressed.node = NULL;
+		seat->pressed.surface = NULL;
+	}
```

Comment:
We have a `seat_reset_pressed(struct seat *seat)` now, that one also handles removing the additional destroy handler set up there.

Reply:
Sorry - forgot. Okay will drop this commit.

Reply:
I think the commit is very valid, would just need to call the `seat_reset_pressed()` wrapper instead of manually resets.